sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -305,7 +305,8 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
   }
 
   private def findBloomFilterWithKey(plan: LogicalPlan, key: Expression): Boolean = {
-    plan.exists {
+    // Ensure that sibling nodes under the same Join do not have runtime filter
+    !plan.exists(_.isInstanceOf[Join]) && plan.exists {
```

Comment:
it looks also wrong to skip the check completely if the plan contains Join.

Reply:
Now, the support are multiple levels. If there are two nested joins that require shuffling, then when transforming up, runtime filter will first be inserted at one end of the lower level join. When scrolling up to the upper join, the lower runtime filter can often be inserted into one end of the upper join.

For example, `bigA join B join BigC where a=b and b=c` cannot ignore the opportunity to insert runtime filter into BigC just because B has inserted a runtime filter into bigA.